### PR TITLE
Support docker for gitian builds

### PR DESCRIPTION
**Pull request by TheCharlatan**
Building with docker is arguably easier and more familiar to most people
than either kvm, or lxc.

This commit also relaxes the back compat requirement a bit. 32 bit linux
now uses glibc version 2.0. Also, the docker shell could not handle gcc arguments
containing spaces, so the explicit '-DFELT_TYPE' declaration was dropped.

Remove some packages from the osx descriptor.

To quickly test this on linux, install docker on a debian-like system (needs apt-get) and create a fresh `gitianuser` user. Make sure the user has full permissions to docker, or run the build script with `sudo`. Then clone my fork of monero: `git clone git@github.com:TheCharlatan/monero`. Copy the `gitian-build.py` script to the gitianuser's home directory and run:

```bash
./gitian-build.py --setup --docker --url https://github.com/TheCharlatan/monero testuser 4a9257b46438b5fb8fe5d58ee8c3b2bbe2b826cc
./gitian-build.py --build --no-commit --detach-sign --docker --url https://github.com/TheCharlatan/monero testuser 4a9257b46438b5fb8fe5d58ee8c3b2bbe2b826cc
```

### CODE REVIEW

File: contrib/gitian/README.md [hunk 1]
```diff
@@ -13,6 +13,10 @@ Multiple developers build the source code by following a specific descriptor
 These results are compared and only if they match, the build is accepted and provided
 for download.
 
+Gitian runs compilation steps in an insolated container. It is flexible and gives you full
```
Comment (moneromooo-monero) [inline contrib/gitian/README.md]: isolated ?

File: contrib/gitian/README.md [hunk 2]
```diff
@@ -63,27 +73,45 @@ reboot
...
+```
+
 
 Manual and Building
 -------------------
-The instructions below use the automated script [gitian-build.py](https://github.com/betcoin/bitcoin/blob/master/contrib/gitian-build.py) which only works in Ubuntu. 
-It calls all available descriptors. Help for the build steps taken can be accessed with `./gitian-build.py --help`.
+
+The instructions below use the automated script [gitian-build.py](gitian-build.py) which only works in Ubuntu. 
+It calls all available .yml descriptors, which in turn pass the build configurations for different plattforms to gitian.
```
Comment (moneromooo-monero) [inline contrib/gitian/README.md]: platforms

File: contrib/gitian/README.md [hunk 3]
```diff
@@ -63,27 +73,45 @@ reboot
...
 Setup the required environment, you only need to do this once:
 
-```
+```bash
 ./gitian-build.py --setup fluffypony v0.14.0
 ```
 
-Where `fluffypony` is your Github name and `v0.14.0` is the version tag you want to build.
+Where `fluffypony` is your Github name and `v0.14.0` is the version tag you want to build. 
+If you are using docker has a host, run it with:
```
Comment (moneromooo-monero) [inline contrib/gitian/README.md]: as a host ?

File: contrib/gitian/README.md [hunk 4]
```diff
@@ -13,6 +13,10 @@ Multiple developers build the source code by following a specific descriptor
 These results are compared and only if they match, the build is accepted and provided
 for download.
 
+Gitian runs compilation steps in an insolated container. It is flexible and gives you full
+control on the build environment, while still ensuring reproducibility and
```
Comment (moneromooo-monero) [inline contrib/gitian/README.md]: over the build...

**Comment by EmbeddedAndroid** [comment]
Thanks for this PR. I just tried this out, and it immediately tried to install docker.io when I had docker-ce installed already. Would be nice to check if docker is already installed, rather than brute forcing an install.

**Pull request merged**